### ISSUE TITLE
accuracy, loss and validation accuracy is nan for a simple neural network model

### ISSUE THREAD

**Issue by vibrancy** [keras-team/keras#14215]
Data: https://www.dropbox.com/s/3mqdgfkz7fjtlf8/titanic_all_numeric.csv?dl=0

```
import tensorflow as tf
from tensorflow import *
from tensorflow.keras.callbacks import EarlyStopping 
from tensorflow.keras.utils import to_categorical
### data
tnc_data = pd.read_csv('https://www.dropbox.com/s/3mqdgfkz7fjtlf8/titanic_all_numeric.csv?dl=1')
### data prep
X = tnc_data.drop( 'survived', axis= 1 ).to_numpy()
X = np.asarray(X).astype(np.float32)
Y = to_categorical( tnc_data['survived'] )

tf.keras.backend.clear_session()
e_stop_monitor = EarlyStopping(patience= 10)

### model 
model1 = keras.Sequential()
model1.add( keras.layers.Dense( units= 10, activation= 'relu', input_shape= (X.shape[1],)  ) )
model1.add( keras.layers.Dense( units= 10, activation= 'relu' ) )
model1.add( keras.layers.Dense( units= 2, activation= 'softmax' ) )
model1.compile( optimizer= 'adam', loss= 'categorical_crossentropy', metrics= ['accuracy'] ) 
model1_training = model1.fit( x= X, y= Y, epochs= 15,
                              validation_split = 20./100,
                              callbacks= [e_stop_monitor],
                              verbose= 2 )

```
When I train the model,  the validation accuracy is improving but accuracy, loss and val_loss are mostly _nan_ as shown below:

```
Epoch 1/15
23/23 - 0s - accuracy: 0.4823 - loss: nan - val_accuracy: 0.4637 - val_loss: nan
Epoch 2/15
23/23 - 0s - accuracy: nan - loss: nan - val_accuracy: 0.6425 - val_loss: nan
Epoch 3/15
23/23 - 0s - accuracy: nan - loss: 0.0000e+00 - val_accuracy: 0.6425 - val_loss: nan
Epoch 4/15
23/23 - 0s - accuracy: nan - loss: nan - val_accuracy: 0.6425 - val_loss: nan
Epoch 5/15
23/23 - 0s - accuracy: 0.5471 - loss: nan - val_accuracy: 0.6425 - val_loss: nan
Epoch 6/15
23/23 - 0s - accuracy: nan - loss: nan - val_accuracy: 0.6425 - val_loss: nan
Epoch 7/15
23/23 - 0s - accuracy: nan - loss: nan - val_accuracy: 0.6425 - val_loss: nan
Epoch 8/15
23/23 - 0s - accuracy: 0.5941 - loss: nan - val_accuracy: 0.5307 - val_loss: nan
Epoch 9/15
23/23 - 0s - accuracy: nan - loss: nan - val_accuracy: 0.5196 - val_loss: nan
Epoch 10/15
23/23 - 0s - accuracy: 0.6296 - loss: nan - val_accuracy: 0.5978 - val_loss: nan
```

**Comment by vibrancy** [comment]
Keras version: '2.4.0'
TensorFlow version: '2.3.0'
OS: Ubuntu 20.04.1 LTS
GPU: Radeon RX 580 Series
Python running in Jupyter Notebook

Thanks in advanced for your help :)

**Comment by PRT28** [comment]
I want to work over this issue

**Comment by nishchalnishant** [comment]
![image](https://user-images.githubusercontent.com/40529054/94165876-4becea80-fea8-11ea-932e-a38479587f1f.png)

your code ran well on colab , i think the problem is from your end , 
>>> try to rerun the code after closing jupyter notebook instance
>>>try to execute the code after creating a new environment  and fresh installation 

![image](https://user-images.githubusercontent.com/40529054/94166581-1ac0ea00-fea9-11ea-8a06-5f56bc488e46.png)

**Comment by vibrancy** [comment]
Thanks Nsichal for running and verifying that it works. I even tried uninstalling and reinstalling Jupiter, Tensorflow ROCM libraries, even reinstalling Ubuntu for god sake!! Then in some forum in _remote part of the internet_ I read that this could be due to ROCM library. Then I tried in Google Colab, just ran with CPU only and voila it ran!

So there is some issue with AMD ROCM tensorflow library! I returned my AMD GPU and installed NVIDIA GEFORCE RTX 2060 super. And the training model ran smoothly....
